### ISSUE TITLE
Error when fetching messages from API

### ISSUE THREAD

**Issue by iammichiel** [esendex/esendex-php-sdk#18]
Hello, 

When calling the API in order to get new messages from inbox, we have the following error : 
```
characterSet() value was '' and must be either 'GSM' or 'Unicode'
```

We made no code change, this simply stopped working. 
Are you aware of a potential bug on your side?

**Comment by iammichiel** [comment]
And the problem was caused by a corrupt message on the server-side of the API : 

```
<messagebody xmlns="http://api.esendex.com/ns/">
<bodytext>
027000002A151221004100000000000002DF00BFB6A5EC1B0FDCFA80F040FF0FA00000001851020000000230505348
</bodytext>
</messagebody>
```

**Comment by iammichiel** [comment]
The same error has occured two days later. Is there any chance you could look in to this ?

**Comment by moretonb** [comment]
Thank you for bringing this to our attention Michiel, apologies for the delay in a response. I'll look into the circumstances that are causing this issue and follow it up.

**Comment by moretonb** [comment]
Morning Michiel,

Using the InboxService I can't see where it would load a body? Are you making a subsequent call to the MessageBodyService?

If that is the case I need a little more information, would you be able to tell me the received date/time for the offending message?

**Comment by iammichiel** [comment]
Hello, 

Yes, we are indeed calling the `MessageBodyService` foreach new incoming message. Here is the incriminated code : 

```php
$bodyService = new MessageBodyService($this->getAuthentication());
$service = new InboxService($this->getAuthentication());
$messages = $service->latest(0, 100);

foreach ($messages as $message) {
    $body = $bodyService->getMessageBodyById($message->id())->bodyText();
}
```

I have deleted the message since. I will slightly adapt the code to allow failure with a try/catch and leave the message in the inbox next time.

**Comment by moretonb** [comment]
I see, do you have a date/time for the message that was causing you an issue available somewhere?

If so I will be able to diagnose the cause further so that we can make any fixes needed to either the SDK or elsewhere.

**Comment by iammichiel** [comment]
I don't remember the exact timing but it occurred the 12th june between 12:00 and 13:00. (Paris time)

**Comment by moretonb** [comment]
Cool, thank you Michiel. I'll see what I can find and I'll update this issue with any outcomes.

**Comment by moretonb** [comment]
I've managed to isolate and replicate your issue. I've raised a bug card in our backlog to be prioritised. Thank you again Michiel.

**Comment by SLStinson** [comment]
Hi Michiel. We have just released a new version of the SDK (1.1.2) which has a fix for this issue. You will now receive a character set of "None" for messages which don't have a character set. 
I'm going to close this issue, but if you have any further issues feel free to let us know and we will look into them as soon as possible. 

Thanks,
Steven.
